sdks/router-sdk/src/entities/mixedRoute/route.ts

```diff
@@ -54,17 +55,22 @@ export class MixedRouteSDK<TInput extends Currency, TOutput extends Currency> {
       const prevPool = pools[i - 1]
       const pool = pools[i]
       const inputToken = tokenPath[i]
+      const isWrappedNativeEquality = ((pool.token0.equals(inputToken.wrapped) || pool.token1.equals(inputToken.wrapped)) && inputToken.isNative)
+      || ((pool.token0.wrapped.equals(inputToken) && pool.token0.isNative) || (pool.token1.wrapped.equals(inputToken) && pool.token1.isNative))
 
       const outputToken =
         pool instanceof V4Pool
-          ? pool.token0.equals(inputToken)
+          ? (isWrappedNativeEquality
```

Comment:
ok just a bit confused here... so even without any changes, why does this special case v4 pools? on non v4 pools it checks the wrapped objects but why again?

and then these new lines I'm confused about... in the isWrappedNativeEquality case it could either be that the inputToken is eth and so we need to assign the output to the non wrapped token in the pool OR that the inputToken is weth so we need to assign the output to the non native token. How does this encapsulate those cases?